src/flint/fmpz.jl

```diff
@@ -55,6 +55,18 @@ base_ring(a::ZZRing) = Union{}
 
 is_domain_type(::Type{ZZRingElem}) = true
 
+###############################################################################
+#
+#   Internal data
+#
+###############################################################################
+
+const ZZRingElemOrPtr = Union{ZZRingElem, Ref{ZZRingElem}, Ptr{ZZRingElem}}
```

Comment:
We need such types for all ring elem and matrix elem subtypes, and ideally in a central places (say in FlintTypes.jl).

Reply:
I am working on a semi-automatic way to generate them

Reply:
It may also be useful to have `const FlintTypes = Union{...}` of all types provided by FLINT/Nemo (or maybe `NemoTypes` is the better name?)

Reply:
> I am working on a semi-automatic way to generate them

See https://github.com/Nemocas/Nemo.jl/pull/1813